doc/HOWTO_RELEASE.rst.txt

```diff
@@ -18,16 +18,16 @@ Source tree
 NumPy Docs
 ----------
 * https://github.com/numpy/numpy/blob/master/doc/HOWTO_RELEASE.rst.txt
-* http://projects.scipy.org/numpy/wiki/MicrosoftToolchainSupport
+* http://projects.scipy.org/numpy/wiki/MicrosoftToolchainSupport (dead link)
```

Comment:
Not sure this is super useful, but I suppose it at least identifies the issues for a future PR

Reply:
Or, if anyone wants to suggest a better link, no better time than now.

Reply:
This [email exchange](http://scipy-user.10969.n7.nabble.com/Building-extensions-on-Win64-with-GNU-compilers-td3156.html) suggests it should be removed

Reply:
All the dead links as well as references to SourceForge can be removed.